internal/services/compute/managed_disk_resource.go

```diff
@@ -298,12 +306,25 @@ func resourceManagedDiskCreate(d *pluginsdk.ResourceData, meta interface{}) erro
 	if createOption == compute.DiskCreateOptionFromImage {
 		imageReferenceId := d.Get("image_reference_id").(string)
 		if imageReferenceId == "" {
-			return fmt.Errorf("`image_reference_id` must be specified when `create_option` is set to `Import`")
+			return fmt.Errorf("`image_reference_id` must be specified when `create_option` is set to `FromImage`")
 		}
 
-		props.CreationData.ImageReference = &compute.ImageDiskReference{
+		imageDiskReference := &compute.ImageDiskReference{
 			ID: utils.String(imageReferenceId),
 		}
+
+		imageReferenceLun := d.Get("image_reference_lun").(int)
+		if imageReferenceLun != -1 {
+			imageDiskReference.Lun = utils.Int32(int32(imageReferenceLun))
+		}
+
+		if _, err := parse.SharedImageVersionID(imageReferenceId); err == nil {
+			log.Printf("[INFO] image_reference_id is a gallery image version id, copying the disk from gallery image.")
+			props.CreationData.GalleryImageReference = imageDiskReference
+		} else {
+			log.Printf("[INFO] image_reference_id is not recognized as a galley image version id, copying the disk from platform/marketplace image.")
+			props.CreationData.ImageReference = imageDiskReference
+		}
```

Comment:
rather than reusing the existing Property here, we should introduce a new one for this field

Reply:
Split it into two properties to make the property meaning clear